### PR TITLE
Doc: Add 'how to' for running a local store

### PR DESCRIPTION
**Pull request by lachypjones**
# Description

# Self-review quick check

 * [x] Make decisions that cost a lot to reverse explicit in the PR description.
 * [x] Avoid nested conditions.
 * [x] Delete dead code and redundant comments.
 * [x] Normalise symmetries by sticking to doing identical things identically. 
 ```
 // one way to handle errors
 if err := f(); err != nil {
    ...
 }
 
 // one way to handle multiple returns
 val, err := f()
 if err != nil {
    ...
 }
 ...
 ```
 * [x] Check that coupled code elements, files, and directories are adjacent. For example, test data is stored as close as possible to a test.
 * [x] Put variable declaration and initialisation together.
 * [x] Divide large expressions into digestable and self-explanatory ones. Use multiple variables if required.
 * [x] Put a blank line between two logically different chunks of code.

### CODE REVIEW

File: docs/how-to/run-local-store.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+convension is required
+
+.. code-block:: console
+    $ mkdir -p fake-store/sdk-store/<sdk>/<release>/<channel>
+
+Where:
+- <sdk> is the sdk name (ie. my-sdk)
+- <release> is the sdk release (ie. latest)
+- <channel> is the sdk channel (ie. edge)
+
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: double empty line before heading titles

File: docs/how-to/run-local-store.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Copying the SDK
+---------------
+
+The SDK should be placed in the most nested directory created above
+(ie. fake-store/sdk-store/my-sdk/latest/edge/my-sdk) and the
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: use `:file:` for paths; end directories with a slash for visibility:
```rst
:file:`/path/foo/bar/dir/` 
```

File: docs/how-to/run-local-store.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+corresponding sdk definition (ie. my-sdk.yaml) should be renamed
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]:
```suggestion
corresponding SDK definition (i.e. :file:`my-sdk.yaml`) should be renamed
```

File: docs/how-to/run-local-store.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+to sdk.yaml and placed in the same location.
+
+.. code-block:: console
+    $ ls fake-store/sdk-store/my-sdk/latest/edge
+    my-sdk.sdk  sdk.yaml
+
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: double blank line

File: docs/how-to/run-local-store.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+Reverting changes
+-----------------
+
+To go back to the default store, simply execute:
+
+.. code-block:: console
+    $ sudo snap set workshop store.url=""
+
+Workshop will now use the default URL
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: period at the end.

File: docs/how-to/run-local-store.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+.. code-block:: console
+    $ sudo snap set workshop store.url=http://localhost:8080/storage/v1/
+    $ sudo snap restart workshop
+
+That's it! Workshop will now pull from your local store
+
+Reverting changes
+-----------------
+
+To go back to the default store, simply execute:
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: down with buzzwords

```suggestion
To go back to the default store:
```

File: docs/how-to/run-local-store.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+Using the local store with Workshop
+-----------------------------------
+
+Override the workshop URL:
+.. code-block:: console
+    $ sudo snap set workshop store.url=http://localhost:8080/storage/v1/
+    $ sudo snap restart workshop
+
+That's it! Workshop will now pull from your local store
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: period, double blank line

File: docs/how-to/run-local-store.rst [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+.. code-block:: console
+    $ ls fake-store/sdk-store/my-sdk/latest/edge
+    my-sdk.sdk  sdk.yaml
+
+Running the local store
+-----------------------
+
+.. code-block:: console
+    $ go run github.com/fsouza/fake-gcs-server@latest -data <path-to-dir-containing-sdk-store> -filesystem-root <same-as-data> -scheme http -port 8080 -public-host localhost:8080
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: line-wrap command sample

File: docs/how-to/run-local-store.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+---------------
+
+The SDK should be placed in the most nested directory created above
+(ie. fake-store/sdk-store/my-sdk/latest/edge/my-sdk) and the
+corresponding sdk definition (ie. my-sdk.yaml) should be renamed
+to sdk.yaml and placed in the same location.
+
+.. code-block:: console
+    $ ls fake-store/sdk-store/my-sdk/latest/edge
+    my-sdk.sdk  sdk.yaml
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: separate output like this (indent to the starting character of the command):

```suggestion
.. code-block:: console
    $ ls fake-store/sdk-store/my-sdk/latest/edge

      my-sdk.sdk  sdk.yaml
```

File: docs/how-to/run-local-store.rst [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+Where:
+- <sdk> is the sdk name (ie. my-sdk)
+- <release> is the sdk release (ie. latest)
+- <channel> is the sdk channel (ie. edge)
+
+Copying the SDK
+---------------
+
+The SDK should be placed in the most nested directory created above
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: be a bit more specific and imperative:

```suggestion
Place the SDK under the channel directory created at the previous step.
```

or 

```suggestion
Place the SDK in the deepest directory created at the previous step.
```

File: docs/how-to/run-local-store.rst [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+You can call the 'fake-store' directory whatever you
+wish, however the remaining structure and naming
+convension is required
+
+.. code-block:: console
+    $ mkdir -p fake-store/sdk-store/<sdk>/<release>/<channel>
+
+Where:
+- <sdk> is the sdk name (ie. my-sdk)
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: use `:samp:` for placeholders and literal snippets, `:file:` for file/directory names
```suggestion
- :samp:`<sdk>` is the SDK name (ie. :samp:`my-sdk`)
```
etc.

File: docs/how-to/run-local-store.rst [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+    This guide assumes that you are familiar with SDKcraft and have
+    a target SDK in mind.
+
+
+Create the directory structure
+------------------------------
+
+The SDK store uses directory structure to determine
+sdk names and channels. Because of this, when running
```
Comment (akcano) [inline docs/how-to/run-local-store.rst]: globally:

```suggestion
SDK names and channels. Because of this, when running
```

**Review by akcano — CHANGES_REQUESTED** [review]
Great job, thanks! Left a few comments, will do a full review when they're implemented.

**Comment by dmitry-lyfar** [comment]
@lachypjones @akcano thanks, looks good! One thing.

This should be part of the contributing document for developers as this is a mock (similar to how snapd organises its e2e testing that involves a store); this is something I'd like to have in the development documentation only (for testing purposes). Once the actual store is up, Workshop will work only with the Store provided by Canonical that is not possible to run self-hosted.

@lachypjones could you please move it there?

File: docs/contributing.rst [hunk 13]
```diff
@@ -206,6 +206,152 @@ of ``Spread``:
...
+corresponding SDK definition (ie. my-sdk.yaml) should be renamed
+to sdk.yaml and placed in the same location.
+
+.. code-block:: console
+
+   $ ls fake-store/sdk-store/my-sdk/latest/edge
+
+     my-sdk.sdk  sdk.yaml
+
+
```
Comment (akcano) [inline docs/contributing.rst]: Not sure where the duplication came from. Drop it, please, and we're done here. :)

```suggestion
```

**Review by akcano — CHANGES_REQUESTED** [review]

**Comment by akcano** [comment]
@dmitry-lyfar I'm entirely fine with having this as part of `CONTRIBUTING` for now, but this raises the question of whether `CONTRIBUTING` is for people who want to contribute to Workshop or for people who work on SDKs, because the subject matter itself crosses the line between Workshop and SDKcraft. 

The only reason I didn't suggest moving it to `CONTRIBUTING` in SDKcraft docs is that this relies on Workshop implementation details that would be out of place there.

**Comment by dmitry-lyfar** [comment]
@akcano this was mostly done for developers who work on Workshop as they need to put up a local store for faster/easier testing locally (versus running a spread test that does it in prepare). So this is why we put it here, in the Workshop contributing guide. SDKcraft developers do not interact with a local store now.

**Review by akcano — APPROVED** [review]

**Review by dmitry-lyfar — APPROVED** [review]

**Pull request merged**